### ISSUE TITLE
Loading Progress

### ISSUE THREAD

**Issue by Zumbalamambo** [isl-org/Open3D#2800]
I'm using the following code to load the 3d model,

```
    def load(self, path):
        self._scene.scene.clear_geometry()
        geometry = None
        geometry_type = o3d.io.read_file_geometry_type(path)

        mesh = None
        if geometry_type & o3d.io.CONTAINS_TRIANGLES:
            mesh = o3d.io.read_triangle_mesh(path)
        if mesh is not None:
            if len(mesh.triangles) == 0:
                print(
                    "[WARNING] Contains 0 triangles, will read as point cloud")
                mesh = None
            else:
                mesh.compute_vertex_normals()
                if len(mesh.vertex_colors) == 0:
                    mesh.paint_uniform_color([1, 1, 1])
                geometry = mesh
            # Make sure the mesh has texture coordinates
            if not mesh.has_triangle_uvs():
                uv = np.array([[0.0, 0.0]] * (3 * len(mesh.triangles)))
                mesh.triangle_uvs = o3d.utility.Vector2dVector(uv)
        else:
            print("[Info]", path, "appears to be a point cloud")
            mesh = None

        if geometry is None:
            ioProgressAmount = 0.5
            cloud = None
            try:
                cloud = o3d.io.read_point_cloud(path)
            except:
                pass
            if cloud is not None:
                print("[Info] Successfully read", path)
                if not cloud.has_normals():
                    cloud.estimate_normals()
                cloud.normalize_normals()
                geometry = cloud
            else:
                print("[WARNING] Failed to read points", path)
                cloud = None

        if geometry is not None:
            self._scene.scene.add_geometry("__model__", geometry,
                                           self.settings.material)
            bounds = geometry.get_axis_aligned_bounding_box()
            self._scene.setup_camera(60, bounds, bounds.get_center())
```

How do I retrieve the progress percentage while the model is being loaded or when it's complete?

**Comment by prewettg** [comment]
Right now the progress percentage is not actually being returned (except I think for PLY files). This is a regression that happened when we switched to using the Assimp library to load files. It is on our list of bugs and we hope to get to it soon.

**Comment by theNded** [comment]
Should have been fixed in #3228. Closed for now.
